src/vm/siginfo.cpp

```diff
@@ -4953,11 +4953,54 @@ void PromoteCarefully(promote_func   fn,
     (*fn) (ppObj, sc, flags);
 }
 
+void ReportByRefPointersFromByRefLikeObject(promote_func *fn, ScanContext *sc, PTR_MethodTable pMT, PTR_VOID pSrc)
+{
+    WRAPPER_NO_CONTRACT;
+
+    _ASSERTE(pMT->IsByRefLike());
+
+    if (pMT == g_TypedReferenceMT)
```

Comment:
It looks to me like outermost call to `ReportByRefPointersFromByRefLikeObject` would iterate over the first field and recursively call itself recursively for `field1`, where it would would return because it's a `TypedReference`. The outermost call to `ReportByRefPointersFromByRefLikeObject` would then recursively call itself for `field2`, where it would report the `ByReference` inside the `Span<T>`. Then the outermost call to `ReportByRefPointersFromByRefLikeObject` would recursively call itself for `field3`, which would return because it's a TypedReference. Did I miss something?